discovery-provider/src/tasks/users.py

```diff
@@ -513,6 +518,19 @@ def update_user_events(
         for event, value in events.items():
             if event == "referrer" and isinstance(value, int):
                 user_events.referrer = value
+                bus.dispatch(
```

Comment:
Just adding a comment here to make sure we adapt this to your other PR with the event flushing